src/pages/api/loadQuestionList.ts

```diff
@@ -5,16 +5,35 @@ import { ID } from 'src/types/common'
 export interface LoadQuestionListParams {
   cid: ID
   topic?: string
+  page: number
 }
 
 export interface LoadQuestionListResults {
   problemList: QStem[]
+  maxPages: number
```

Comment:
did you refer to any pre-existing pagination API? One that I know is:

```typescript
interface Params {
    baseQuestionId: ID | null
    questionNum: number
}

interface Results {
    questions: QStem[]
    next: ID | null
    prev: ID | null
}
```

Reply:
> did you refer to any pre-existing pagination API? One that I know is:
> 
> ```ts
> interface Params {
>     baseQuestionId: ID | null
>     questionNum: number
> }
> 
> interface Results {
>     questions: QStem[]
>     next: ID | null
>     prev: ID | null
> }
> ```

I tried to refer to them but they don't seem to align with our needs since we have to filter responses based on cid and topic and also we will have to get page number assuming there is a large number of questions added. 

Should I update it this way? ... I think only having next and previous button on pagination might not be ideal assuming there is large number of questions leading to large number of pages. 🤔

Reply:
oh the example is just a general interface. We should add `cid` and `topic` for sure.

Reply:
> oh the example is just a general interface. We should add `cid` and `topic` for sure.

I see should we stick with this one?

Reply:
I think we can stick to what we have rn (cuz it works). Let's focus on something more important. I just brought this up to encourage you to think more generalizable interface:)